src/init/starship.ps1

```diff
@@ -136,22 +136,37 @@ $null = New-Module starship {
         $arguments += "--status=$($lastExitCodeForPrompt)"
 
         # Invoke Starship
-        $promptText = if ($script:TransientPrompt) {
+        $promptText, $rpromptText = $null, $null
+        if ($script:TransientPrompt) {
             $script:TransientPrompt = $false
-            if (Test-Path function:Invoke-Starship-TransientFunction) {
+            $promptText = if (Test-Path function:Invoke-Starship-TransientFunction) {
                 Invoke-Starship-TransientFunction
             } else {
                 "$([char]0x1B)[1;32m❯$([char]0x1B)[0m "
             }
+            $rpromptText = if (Test-Path function:Invoke-Starship-TransientRightFunction) {
+                Invoke-Starship-TransientRightFunction
+            } else {
+                ""
+            }
         } else {
-            Invoke-Native -Executable ::STARSHIP:: -Arguments $arguments
+            $promptText = Invoke-Native -Executable ::STARSHIP:: -Arguments $arguments
+            $arguments += "--right"
+            $rpromptText = Invoke-Native -Executable ::STARSHIP:: -Arguments $arguments
```

Comment:
Looking at this, I should check if there's a notable performance difference between both approaches.

Reply:
This approach is about 20ms slower for a minimal rprompt on my Windows system, which makes me somewhat more inclined to go with https://github.com/starship/starship/pull/4208, given that all users will feel this slowdown.

Reply:
I mean yes, two starship invocations will definitely be more time-consuming than one. 

I think we can still go with this for now, and when #3467 finally lands, we could go with the approach in #4208 as a performance improvement.

Reply:
Is there an easy method to spawn both executables concurrently?

Reply:
There are ways using Jobs and Threads, but they're ugly (see https://stackoverflow.com/questions/4016451/can-powershell-run-commands-in-parallel) and I'm not sure the overhead of these would justify their usage here.